docs/data/data-grid/sorting/sorting.md

```diff
@@ -71,6 +71,14 @@ In the following demo, the user cannot use the _rating_ column as a sorting rule
 
 {{"demo": "DisableSortingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
+### Read-only sorting
```

Comment:
Couldn't think of a more appropriate name for this, technically it _is_ read-only but from the perspective of the grid's default UI, it could be implemented on the developer's side to be _not-read-only_, like in the example below.

Reply:
Yeah I also don't like the name, I think it makes it sound more complicated than it is. What we want to say is that the `sortable` flag controls whether or not the sorting is enabled for the end-user from the UI. I think I would integrate that information to the paragraph above, in `Disable the sorting`, and maybe merge both examples together.

Reply:
In the context of sorting only, I couldn't agree more, but after working on https://github.com/mui/mui-x/pull/11538 it seemed to be more than just disabling a column filter. Considering this example https://codesandbox.io/p/sandbox/bold-platform-9jg53c?file=%2Fsrc%2FDemo.tsx%3A41%2C9
1. The filters will be there in the filter panel with a slightly different experience (disabled/read only) so technically it _will_ impact the default Grid UI
2. It could be a genuine use-case for some users, imagine some applied filters some of which are not allowed to be changed by a specific user type.
3. Putting it under a separate heading may improve the discoverability and present this as a concept.

So keeping in mind the behavior for filtering, I thought it'd be better to present this as a concept (read-only object) for other flags too, that way it will be more:
- Consistent in different flags (like groupable, pinnable, filterable, sortable, ...)
- Discoverable
- Help support some specific use cases like #8357 (which could be true even without it but more discoverability may help find it easily)

Would appreciate your thoughts on this.

Reply:
I've read again and I still feel like it should be in the same paragraph as `Disable the sorting`. Because both paragraph describe the same thing: the behavior of the `.sortable` flag. They both talk about what happens when `sortable: false`. Splitting it into a different paragraph makes it somewhat easier to discover that case if one is looking very specifically for that, but it splits the information about the behavior of the flag into two paragraphs/sections.

It's a soft disagree though, so if that doesn't convince you let's leave it as it is.

Reply:
Valid point, I think I also seem to agree with you on this, essentially it's disabling the column filter, and it's necessary to explain the behavioral aspect of disabling in the disabling section too, like I [did](https://deploy-preview-11574--material-ui-x.netlify.app/x/react-data-grid/aggregation/#for-some-columns) for the aggregation. I'll update it for the others too.

Reply:
How about "Sorting non-sortable columns programmatically" instead of "Read-only sorting"?

Reply:
Cool, now there are two options here.
1. Make it [a sub-section](https://deploy-preview-11512--material-ui-x.netlify.app/x/react-data-grid/sorting/#disable-the-sorting) (or part of it, without having a heading) of `Disable the ${featureName}` > `For columns`.
2. Keep it a separate section, put it on the same nested level as `Disable the ${featureName}` but name it more generalized (like @cherniavskii suggested) for better discoverability.

I also noticed we miss a global disabling prop (like disableColumnSort) for sorting like we have for every other feature (e.g. disableColumnFilter, disableColumnMenu, disableVIrtualization, ...), that is why we don't have a document structure like [this](https://mui.com/x/react-data-grid/aggregation/#disable-aggregation) (`For all columns` and `For some columns` nested levels) in sorting.

I think we should add this flag to the sorting and if it is missing elsewhere (I have to check) and then make documentation for every object like so:

```js
### Disable ${objectName}

#### For all columns

content...

#### For some columns

content...

#### `${featureTitle}able non-${featureTitle}able columns programmatically`

content...

```

This way it will be:
1. A sub-section of the `Disable` section (@romgrk's point)
2. A separate section for this specific thing (better for SEO and discoverability)
3. It will be consistent and aligned in all the objects, thus could be mentioned as a central behavior in docs pages like FAQs.

Any objections?

Reply:
Another related question, which applies to all the features, should we allow programmatically sorting some columns if the sorting feature is disabled altogether (using the proposed prop `disableColumnSorting`, I don't think so, wdyt?
@romgrk @cherniavskii

Reply:
No objection to the first comment.

For the behavior of the `disableColumnXxxx` flags, I'm not sure what's their use-case. Without context I'd argue that they should control the UI and not disable programmatic actions, just like the column-local flags. But depending on their historic use I could change my opinion. As long as they're all consistent I'm happy.

Reply:
> For the behavior of the `disableColumnXxxx` flags, I'm not sure what's their use-case. 

As how I understand these flags, these refer to disabling the entire feature. Like how the term implies, for instance `disableColumnFiltering`, the most accurate understanding I can derive from this is turning the entire filtering feature off. That is the reason I suggested not to consider supporting programmatic actions for such features.

Reply:
I was a bit mistaken on this, `disableColumnFilter` [_allows_](https://codesandbox.io/p/sandbox/boring-torvalds-fc7lf3?file=%2Fsrc%2FDemo.tsx%3A28%2C12) to control, initialize or use API method to add filters programmatically. Kept the same behavior on this one too. It only disables stuff on the UI.